ta/arch/arm/user_ta_header.c

```diff
@@ -46,7 +48,21 @@ void __noreturn _C_FUNCTION(__ta_entry)(unsigned long func,
 					struct utee_params *up,
 					unsigned long cmd_id)
 {
-	TEE_Result res = __utee_entry(func, session_id, up, cmd_id);
+	TEE_Result res;
+
+#if defined(CFG_TA_STACK_PROTECTOR)
+	uintptr_t canary;
+
+	res = _utee_cryp_random_number_generate(&canary, sizeof(canary));
+	if (res != TEE_SUCCESS)
+		goto out;
+
+	canary &= ~0xffUL;
```

Comment:
Why? This isn't obvious.

Reply:
This is to clearing a byte in the canary value to prevent string base overflow attack e.g scanf, strcpy. Because it doesn't let you put a null byte in middle of the string. In some rare case, it also prevent leaking the canary value when print non-terminated string from stack. This reduce the entropy but I think 24bits is still enough for 32bits platforms. I will put comment in the patch to make it clear.

Reply:
Makes sense, thanks for the explanation.